google/pubsub_v1/services/publisher/async_client.py

```diff
@@ -315,6 +313,9 @@ async def update_topic(
             ),
         )
 
+        if timeout is None or isinstance(timeout, (int, float)):
+            timeout = timeouts.ConstantTimeout(timeout)
```

Comment:
We shouldn't need except for the `None` case, which is new.  I don't think we should accept `None`.  Apologies if `None` came from me.

`int` and `float` are handled here: 

https://github.com/googleapis/python-api-core/blob/7337c6b123735fb9ae5d0e54b4399275719c020c/google/api_core/gapic_v1/method.py#L66-L67

The same comment applies to other places where you added this.

Reply:
And, of course, not DRY.